### ISSUE TITLE
Mouse pointer "vanishes" when moved over Terminal Window

### ISSUE THREAD

**Issue by inoperable** [microsoft/terminal#18335]
📝where'd this go

_Originally posted by @zadjii-msft in https://github.com/microsoft/terminal/pull/18215#discussion_r1880139859_
            
Probably this

**Comment by zadjii-msft** [comment]
@inoperable Could you add any more details/? That PR merged literally 5 days ago, I'm not sure it's even in a canary build currently

**Comment by zadjii-msft** [comment]
![Image](https://github.com/user-attachments/assets/32714634-1ccf-4ef2-9585-8d617582ba9f)

**Comment by lhecker** [comment]
And please confirm with us what exact version of Windows Terminal you're using.

**Comment by inoperable** [comment]
It's not the behavior intended, I'm very sure about it, the is setting is a good point to eliminate culprits - but itl
What happens is that the pointer gets hidden all the time once the terminal window is in focus and the mouse pointer hovers over it. I can see settings buttons highlighting (as per on mouse over event) but no mouse pointer.

Term is 1.23.241213001-llm, OS 27764.1000 & 19045.4651

**Comment by inoperable** [comment]
And this seems to be only a 27xxx insiders build exklusive, since I can't replicate it with 19045.xxxx 

o.O - so I guess I just wasted your time unnecessary...

**Comment by inoperable** [comment]
It's definitely a insiders build exclusive, I couldn't replicate the same on 19045.4651. I apologize for confusions therefore, maybe this is not entirely a waste of time since this might make a come back agai, I'll fiddle around with the insiders build again and will update this thread once I know more when/how

**Comment by lhecker** [comment]
It's possible that this is a bug across any Windows 11 version, because I did struggle quite a bit with making cursor hiding work right. The problem is that the WinUI APIs for doing so are just kind of bad and not very robust. It would not surprise me if they broke between Windows versions.

**Comment by lhecker** [comment]
Leading theory: The current cursor that we get from the WinUI API is non-null _but also_ hidden. This means that when we restore it, we restore a hidden cursor that won't ever go away, because nothing explicitly sets a visible cursor in our code either. Ideally, WinUI would have a cursor stack that you can push/pop into, but in the absence of that we'll just do the next best thing:
* Remove this code:
  https://github.com/microsoft/terminal/blob/8598ed78d58cf6b10372fca4132bfa9553ac6735/src/cascadia/TerminalControl/TermControl.cpp#L99-L103
* Explicitly restore a visible cursor here:
  https://github.com/microsoft/terminal/blob/8598ed78d58cf6b10372fca4132bfa9553ac6735/src/cascadia/TerminalControl/TermControl.cpp#L79-L80

**Comment by inoperable** [comment]
![Image](https://github.com/user-attachments/assets/7d2be51b-4079-42ac-a74c-e4051b372cdb)
Invisible Mouse™

**Comment by inoperable** [comment]
So, to clarify @lhecker I think you are right, and have it figured out (as mentioned above)

**Comment by Zeroes1** [comment]
@lhecker , @zadjii-msft 

I have another problem (but with cursor of mouse too)

i see this since begin use Windows Terminal terminal-1.17.11461.0 to latest canary release

i see this effect only when use WinSSHTerm + Putty
[https://winsshterm.blogspot.com, https://www.putty.org]

when Alt-Tab from window (WinSSHTerm + Putty) to Windows Terminal window, the mouse cursor set to OS BUSY mode.

No effect when returning from other program. No effect when returning to another window (not WT window).

problem reproduction 100%
open NEW issue for this?
https://github.com/user-attachments/assets/b0170e74-ba67-4323-9244-b766cb9bbe71

**Comment by lhecker** [comment]
Yes, I think that's a separate issue.

**Comment by inoperable** [comment]
W8t,. so what was it?

**Comment by inoperable** [comment]
@lhecker PointerCursor() or it didnt happend

**Comment by lhecker** [comment]
Yeah, I've since reopened https://github.com/microsoft/terminal/issues/18400. The WinRT/WinUI APIs I tried to use for this are simply too busted. I'll try it with basic Win32 code next.

**Comment by inoperable** [comment]
No worries, on the plus side, it makes my mouse battery last longer and invent keybindings to everything ;>

**Comment by inoperable** [comment]
@lhecker have you been able to pinpoint the trigger? I still cant figure out WHAT exactly is causing it. I thought that selection+focus loss the culprit -but nope

**Comment by lhecker** [comment]
As far as I understand it, it's a race condition between our code and WinUI's input stack (running on another thread), which is why it happens never on some setups and quite often on others.

**Comment by Zeroes1** [comment]
my post upper, about mouse busy pointer it's bug in WinSSHTerm. Fixed in v2.41.3

**Comment by inoperable** [comment]
> my post upper, about mouse busy pointer it's bug in WinSSHTerm. Fixed in v2.41.3

how come WinSSHTerm affects not-so-ssh-at-all runtime? As you can see in my screenshots- there is no ssh instance running there
